fastlane/lanes/localization.rb

```diff
@@ -48,4 +69,130 @@
+    FileUtils.cp(RELEASE_NOTES_SOURCE_PATH, File.join(STORE_METADATA_FOLDER, 'default', 'release_notes.txt'))
+
+    # FIXME: Replace this with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
+    target_files = {
+      "v#{release_version_current}-whats-new": { desc: 'release_notes.txt', max_size: 4000 },
+      app_store_name: { desc: 'name.txt', max_size: 30 },
+      app_store_subtitle: { desc: 'subtitle.txt', max_size: 30 },
+      app_store_desc: { desc: 'description.txt', max_size: 4000 },
+      app_store_keywords: { desc: 'keywords.txt', max_size: 100 }
+    }
+    gp_downloadmetadata(
+      project_url: GLOTPRESS_STORE_METADATA_PROJECT_URL,
+      target_files: target_files,
+      locales: GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES,
+      download_path: STORE_METADATA_FOLDER
+    )
+
+    files_to_commit = [File.join(STORE_METADATA_FOLDER, '**', '*.txt')]
+
+    ensure_default_metadata_are_not_overridden(metadata_files_hash: target_files)
+
+    files_to_commit.append(*generate_gitkeep_for_empty_locale_folders)
+
+    git_add(path: files_to_commit, shell_escape: false)
+    git_commit(
+      path: files_to_commit,
+      message: 'Update App Store metadata translations',
+      allow_nothing_to_commit: true
+    )
+
+    sanitize_appstore_keywords
+  end
+
+  # TODO: This ought to be part of the localized metadata download action, or of Fastlane itself.
+  desc 'Updates the files with the localized keywords values for App Store Connect to match the 100 characters requirement'
+  lane :sanitize_appstore_keywords do
+    Dir[File.join(STORE_METADATA_FOLDER, '**')].each do |locale_dir|
+      keywords_path = File.join(locale_dir, 'keywords.txt')
+
+      unless File.exist?(keywords_path)
+        UI.important "Could not find keywords file in #{locale_dir}. Skipping..."
+        next
+      end
+
+      keywords = File.read(keywords_path)
+      app_store_connect_keywords_length_limit = 100
+
+      if keywords.length <= app_store_connect_keywords_length_limit
+        UI.success "#{keywords_path} has less than #{app_store_connect_keywords_length_limit} characters."
+        next
+      end
+
+      UI.important "#{keywords_path} has more than #{app_store_connect_keywords_length_limit} characters. Trimming it..."
+
+      english_comma = ','
+      arabic_comma = '،'
+      locale_code = File.basename(locale_dir)
+      keywords = keywords.gsub(arabic_comma, english_comma) if locale_code == 'ar-SA'
+
+      keywords = keywords.split(english_comma)[0...-1].join(english_comma) until keywords.length <= app_store_connect_keywords_length_limit
+
+      File.write(keywords_path, keywords)
+
+      git_commit(
+        path: keywords_path,
+        message: "Trim #{locale_code} keywords to be less than #{app_store_connect_keywords_length_limit} characters",
+        allow_nothing_to_commit: false
+      )
```

Comment:
I thought it clearer to individually commit each modified file, rather than in a batch or as part of the update-after-download commit. I think it valuable to know when this happen in detail, because it should help us fix the problem at the root.